backend/apps/admin/views.py

```
@@ -0,0 +1,88 @@
+        from datetime import timedelta
+        import calendar
+        
+        days = 30
+        cutoff = timezone.now() - timedelta(days=days)
+        trend = []
+        
+        for i in range(days):
+            day = cutoff + timedelta(days=i)
+            next_day = day + timedelta(days=1)
+            
+            opened = PullRequest.objects.filter(
```

Ruff (0.15.21)

[error] Undefined name `PullRequest`

(F821)